packages/react-flow-editor/src/Editor/helpers/useSelectionZone.ts

```diff
@@ -0,0 +1,53 @@
+import { useStore } from "@nanostores/react"
+import { RefObject, useCallback } from "react"
+
+import { AutoScrollAtom, AutoScrollDirection, DragItemAtom, SelectionZoneAtom, TransformationMap } from "../state"
+import { getRectFromRef } from "./getRectFromRef"
+
+export const useSelectionZone = (zoomContainerRef: RefObject<HTMLElement>) => {
+  const selectionZone = useStore(SelectionZoneAtom)
+
+  const initSelectionZone = useCallback(
+    (e: React.MouseEvent<HTMLElement>) => {
+      if (e.shiftKey) {
+        const transformation = TransformationMap.get()
+        const zoomContainerRect = getRectFromRef(zoomContainerRef)
+
+        const left = (e.clientX - zoomContainerRect.left) / transformation.zoom
+        const top = (e.clientY - zoomContainerRect.top) / transformation.zoom
+        const point = { x: left, y: top }
+
+        SelectionZoneAtom.set({ cornerStart: point, cornerEnd: point })
+      }
+    },
+    [zoomContainerRef]
```

Comment:
Do we really need ref as a deps, it seems that actually `zoomContainerRef.current` changes, but `zoomContainerRef` link is immutable

Reply:
yes, you are right